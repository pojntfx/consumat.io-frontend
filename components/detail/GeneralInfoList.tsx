export type MediaInfo = {
  description: string;
  value: string;
};

type GeneralInfoListProps = {
  infos: MediaInfo[];
  className?: string;
};

const GeneralInfoList = ({ infos, className }: GeneralInfoListProps) => {
  return (
    <div
      className={
        "px-4 pb-4 bg-gradient-to-br from-white to-white dark:from-gray-700 dark:to-gray-800 rounded shadow-md " +
        className
      }
    >
      <h3 className="cardHeading">Infos</h3>
      <div className="flex flex-row">
        {infos.map((info, index) => {
          return (
            <div className="ml-1 mr-4" key={index}>
              <h3>{info.description}</h3>
              <p>{info.value}</p>
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default GeneralInfoList;
