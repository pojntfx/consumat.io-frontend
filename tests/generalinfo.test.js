import React from "react";
import renderer from "react-test-renderer";
import GeneralInfoList from "../components/detail/GeneralInfoList";
import testExamples from "./testExamples";

it("renders correctly", () => {
  const tree = renderer
    .create(<GeneralInfoList infos={testExamples.exampleInfos} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
