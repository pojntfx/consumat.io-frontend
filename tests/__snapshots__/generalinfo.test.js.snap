// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="px-4 pb-4 bg-gradient-to-br from-white to-white dark:from-gray-700 dark:to-gray-800 rounded shadow-md undefined"
>
  <h3
    className="cardHeading"
  >
    INFOS
  </h3>
  <div
    className="flex flex-row"
  />
</div>
`;
